go/staking/api/slashing.go

```diff
@@ -11,20 +13,37 @@ type SlashReason int
 const (
 	// SlashDoubleSigning is slashing due to double signing.
 	SlashDoubleSigning SlashReason = 0
+	SlashMax                       = SlashDoubleSigning
 
-	SlashMax = SlashDoubleSigning
+	SlashDoubleSigningName = "double-signing"
 )
 
 // String returns a string representation of a SlashReason.
 func (s SlashReason) String() string {
 	switch s {
 	case SlashDoubleSigning:
-		return "double-signing"
+		return SlashDoubleSigningName
 	default:
 		return "[unknown slash reason]"
 	}
 }
 
+// MarshalText encodes a SlashReason into text form.
+func (s SlashReason) MarshalText() ([]byte, error) {
+	return []byte(s.String()), nil
```

Comment:
That's not the best way as it could encode as `[unknown slash reason]`. We should probably use a separate switch and return an error in that case?